packages/http/src/router/index.ts

```diff
@@ -14,16 +15,42 @@ import {
 import { matchBaseUrl } from './matchBaseUrl';
 import { matchPath } from './matchPath';
 import { IMatch, MatchType } from './types';
+import { isTemplated } from './matchPath';
+
+// matching score is calcualated for resources based upon their match with given request
+// matchScore is introduced to sort the resources in case we get multiple matches for given request.
+const calculateMatchScore = (endpoint: string, request: string) => {
+  const endpointParts = endpoint.split('/');
+  const requestParts = request.split('/');
+  let score = 0;
+  for (let i = 0; i < endpointParts.length; i++) {
+    if (endpointParts[i] === requestParts[i]) {
+      score++;
+    } else if (isTemplated(endpointParts[i])) {
+      score += 0.5;
```

Comment:
what's the purpose of this score += 0.5; ?

Reply:
Added score to differentiate match weightage. When templated path is found, 0.5 score is added to it and if path is not templated and it is matching with request, then score is incremented by 1. So non-template path will have more weightage compare to templated path while performing sorting.